string name
string stamp
---
bool done